subsys/testsuite/utils/negative_handler.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <ztest.h>
+
+#if defined(CONFIG_TEST_NEGATIVE_FATAL_HANDLER)
+ZTEST_BMEM volatile bool valid_fault;
+
+void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
+{
+	printk("Caught system error -- reason %d %d\n", reason, valid_fault);
+	if (valid_fault) {
+		printk("== fatal error expected as part of test case ==\n");
+		valid_fault = false; /* reset back to normal */
+	} else {
+		printk("== fatal error was unexpected, aborting ==\n");
+		k_fatal_halt(reason);
+	}
+}
+#endif /* CONFIG_TEST_NEGATIVE_FATAL_HANDLER */
+
+
+#if defined(CONFIG_TEST_NEGATIVE_ASSERT_HANDLER)
+ZTEST_BMEM volatile bool valid_assert;
+
+#ifdef CONFIG_ASSERT_NO_FILE_INFO
+void assert_post_action(void)
+#else
+void assert_post_action(const char *file, unsigned int line)
+#endif
+{
+#ifndef CONFIG_ASSERT_NO_FILE_INFO
+	ARG_UNUSED(file);
+	ARG_UNUSED(line);
+#endif
+
+	printk("Caught assert failed\n");
+
+	if (valid_assert) {
+		valid_assert = false; /* reset back to normal */
+	} else {
+		printk("== assert failed was unexpected, aborting ==\n");
```

Comment:
Pleae change it like:
"Assert failure was not expected. Abording....."

Reply:
Thanks for correcting this. Got all the relevant places fixed.